/* Copyright 2019 École Polytechnique Fédérale de Lausanne. All Rights Reserved.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
==============================================================================*/

syntax = "proto3";

package tensorflow;
option cc_enable_arenas = true;
option java_outer_classname = "GateRunnerProtos";
option java_multiple_files = true;
option java_package = "org.tensorflow.framework";

import "tensorflow/core/lib/core/error_codes.proto";

message GateRunnerDef {
    string gate_name = 1;
    repeated string enqueue_op_name = 2;
    string close_op_name = 3;
    string cancel_op_name = 4;
    string device = 5;
    repeated error.Code gate_closed_exception_types = 6;
    bool request_stop_on_success = 7;
}

message CreditSupplierDef {
    string downstream_gate_name = 1;
    string supply_op_name = 2;
    string close_op_name = 3;
    string cancel_op_name = 4;
    string device = 5;
}